torch/nn/_functions/thnn/pooling.py

```diff
@@ -364,6 +364,112 @@ def backward(self, grad_output):
                                                          self.stride[0], self.stride[2], self.stride[1])
         return grad_input
 
+
+class AdaptiveMaxPool1d(Function):
+
+    def __init__(self, output_size, return_indices=False):
+        self.output_size = output_size
```

Comment:
Why did you revert it so that it no longer uses `_single` here? I already said it should be a tuple.